### PR TITLE
Translation for Main Concepts' Handling Events page.

### PR DESCRIPTION
**Pull request by asheerrizvi**
Please point out any grammatical or spelling correction(s) which needs to be made and I'll fix it. Would also appreciate suggestions on words which could be further translated or presented in a better way. 

#22

### CODE REVIEW

File: content/docs/handling-events.md [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
   - "docs/events-ko-KR.html"
 ---
 
-Handling events with React elements is very similar to handling events on DOM elements. There are some syntax differences:
+React اور DOM دونوں میں ایونٹ ہینڈلنگ کی ترتیب ایک دوسرے سے مشابہت رکھتی ہے حالانکہ، syntax میں کچھ چیزیں مختلف ہیں
 
-* React events are named using camelCase, rather than lowercase.
-* With JSX you pass a function as the event handler, rather than a string.
+* React میں ایونٹس کا نام camelCase میں لکھا جاتا ہے بجائے lowercase کے
+* JSX میں آپ فنکشن پاس کریں گے بتور ایونٹ ہینڈلر بجائے ایک string کے
```
Comment (jabranr) [inline content/docs/handling-events.md:14]:
```suggestion
* JSX میں آپ فنکشن پاس کریں گے بطور ایونٹ ہینڈلر بجائے ایک string کے
```

Comment (jabranr) [inline content/docs/handling-events.md:14]: Keep order in sentences

```suggestion
* JSX میں آپ event handler کو string کی بجائے function کے طور پر پیش کرتے ہیں
```

File: content/docs/handling-events.md [hunk 2]
```diff
@@ -54,11 +54,11 @@ function ActionLink() {
 }
 ```
 
-Here, `e` is a synthetic event. React defines these synthetic events according to the [W3C spec](https://www.w3.org/TR/DOM-Level-3-Events/), so you don't need to worry about cross-browser compatibility. See the [`SyntheticEvent`](/docs/events.html) reference guide to learn more.
+یہاں پہ `e` ایک synthethic ایونٹ ہے، ان synthetic ایونٹس کو React نے [W3C spec](https://www.w3.org/TR/DOM-Level-3-Events/) کو مدنظر رکھتے ہوئے بنایا ہے  لہٰذا یہ کراس-براوزر مطابقت رکھتے ہیں ۔ اس موضوع پہ اور معلومات حاصل کرنے کے لیے [`SyntheticEvent`](/docs/events.html) سے منسوب گائیڈ کو ملاحظہ فرمائیں
```
Comment (jabranr) [inline content/docs/handling-events.md:57]:
```suggestion
یہاں پہ `e` ایک synthethic ایونٹ ہے، ان synthetic ایونٹس کو React نے [W3C spec](https://www.w3.org/TR/DOM-Level-3-Events/) کو مدنظر رکھتے ہوئے بنایا ہے  لہٰذا یہ مختلف براوزر مطابقت رکھتے ہیں ۔ اس موضوع پہ مزید معلومات حاصل کرنے کے لیے [`SyntheticEvent`](/docs/events.html) سے منسوب گائیڈ کو ملاحظہ فرمائیں
```

File: content/docs/handling-events.md [hunk 3]
```diff
@@ -54,11 +54,11 @@ function ActionLink() {
...
 
-When using React, you generally don't need to call `addEventListener` to add listeners to a DOM element after it is created. Instead, just provide a listener when the element is initially rendered.
+عموماً React میں DOM کے عناصر پر listeners جوڑنے کے لیے `addEventListener` استعمال کرنے کی ضرورت نہیں پڑتی، یہ کافی ہے کہ جب عناصر render ہو تب آپ listener فراہم کر دیں
 
-When you define a component using an [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes), a common pattern is for an event handler to be a method on the class. For example, this `Toggle` component renders a button that lets the user toggle between "ON" and "OFF" states:
+جب آپ [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) کا استعمال کرتے ہوئے جزو بناتے ہیں تو ایک آم پیٹرن یہ ہے کہ ایونٹ ہینڈلر کو اس class کا طریقہ یا method بنا دیا جائے ۔ مثال کے طور پہ یہ `Toggle` جزو ایک بٹن render کر رہا ہے جس سے ایک یوزر آن اور آف states کے درمیان ٹوگل کر سکتا ہے
```
Comment (jabranr) [inline content/docs/handling-events.md:61]:
```suggestion
جب آپ [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) کا استعمال کرتے ہوئے جزو بناتے ہیں تو ایک عام طریقہ کار یہ ہے کہ ایونٹ ہینڈلر کو اس class کا طریقہ یا method بنا دیا جائے ۔ مثال کے طور پہ یہ `Toggle` جزو ایک بٹن render کر رہا ہے جس سے ایک صارف "ON" اور "OFF" states کے درمیان ٹوگل کر سکتا ہے
```

**Comment by asheerrizvi** [comment]
> There are way too many changes required. This is not a good translation. There are typos, incorrect sentences. I have given some suggestions but I don't want to go through rest of the it unless you go through it first and re-think the translation.

@jabranr Thank you for the feedback, I was unsure about many things so I wanted to put something out first. I will improve on your suggestions and come up with something better.

File: content/docs/handling-events.md [hunk 4]
```diff
@@ -138,11 +138,11 @@ class LoggingButton extends React.Component {
...
+اس syntax کی خامی یہ ہے کہ `LoggingButton` کا ہر ایک render ایک نیا callback بھی بنائے گا، زیادہ تر معاملات میں، یہ ٹھیک ہے۔ حالانکہ اگر اس callback کو ایک prop کے طور پہ نچلے جزو میں پیش کیا جائے تو وہ جزو ایک اضافی re-rendering کر سکتے ہیں ۔ کارکردگی سے متعلق ان مسائل سے بچنے کے لیے ہم آپ کو دو مشورے دیں گے، پہلا constructor میں binding کا استعمال ہے اور دوسرا class fields syntax کا
 
-## ایونٹ ہینڈلرس میں arguments فراہم کرنا {#passing-arguments-to-event-handlers}
+## Event Handlers میں arguments فراہم کرنا {#passing-arguments-to-event-handlers}
 
-لوپ کے اندر ایونٹ ہینڈلر میں ایک اضافی parameter فراہم کرنا ایک آم بات ہے، مثلاً اگر `id` کو رو ID سے نسبت دی جائے تو انمے سے کوئی بھی استعمال میں لایا جا سکتا ہے
+Loop کے اندر event handler میں ایک اضافی parameter فراہم کرنا ایک آم عمل ہے، مثلاً اگر `id` کو  row ID سے نسبت دی جائے تو ان میں سے کوئی بھی طریقہ استعمال میں لایا جا سکتا ہے
```
Comment (jabranr) [inline content/docs/handling-events.md:145]:
```git
- آم
+ عام
```

File: content/docs/handling-events.md [hunk 5]
```diff
@@ -54,11 +54,11 @@ function ActionLink() {
...
 
-عموماً React میں DOM کے عناصر پر listeners جوڑنے کے لیے `addEventListener` استعمال کرنے کی ضرورت نہیں پڑتی، یہ کافی ہے کہ جب عناصر render ہو تب آپ listener فراہم کر دیں
+React میں عموماً DOM کے عناصر پر listeners جوڑنے کے لیے `addEventListener` استعمال نہیں کرنا پڑتا، یہ کافی ہے کہ جب عناصر render ہو تب آپ listener فراہم کر دیں
 
-جب آپ [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) کا استعمال کرتے ہوئے جزو بناتے ہیں تو ایک آم پیٹرن یہ ہے کہ ایونٹ ہینڈلر کو اس class کا طریقہ یا method بنا دیا جائے ۔ مثال کے طور پہ یہ `Toggle` جزو ایک بٹن render کر رہا ہے جس سے ایک یوزر آن اور آف states کے درمیان ٹوگل کر سکتا ہے
+جب آپ [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) کا استعمال کرتے ہوئے جزو بناتے ہیں تو ایک آم طریقہ کار یہ ہے کہ event handler کو اس class کا طریقہ (method) بنا دیا جائے ۔ مثال کے طور پر یہ `Toggle` جزو ایک بٹن render کر رہا ہے جس سے ایک صارف "ON" اور "OFF" states کے درمیان ٹوگل کر سکتا ہے
```
Comment (jabranr) [inline content/docs/handling-events.md:61]:
```
- آم
+ عام
```

**Review by jabranr — CHANGES_REQUESTED** [review]
Great work! Thank you. Just couple of changes for typos.

**Comment by asheerrizvi** [comment]
@jabranr thanks! I have done the requested changes, you can check now.

**Review by jabranr — APPROVED** [review]

**Pull request merged**